src/main/java/io/thinkit/edc/client/connector/services/ApplicationObservability.java

```diff
@@ -46,11 +41,38 @@ public Result<HealthStatus> checkHealth() {
                         .toList();
                 return new Result<>(error);
             }
-        } catch (IOException | InterruptedException | JsonLdError e) {
+        } catch (JsonLdError e) {
             throw new RuntimeException(e);
         }
     }
 
+    public Result<HealthStatus> checkHealth() {
+        try {
+            var requestBuilder = HttpRequest.newBuilder()
+                    .uri(URI.create("%s/check/health".formatted(url)))
+                    .GET();
+
+            var request = interceptor.apply(requestBuilder).build();
+
+            var response = httpClient.send(request, HttpResponse.BodyHandlers.ofInputStream());
+            return getResponse(response);
```

Comment:
these three lines are duplicated for all the sync methods, and also in the async ones the logic is pretty similar.

as this pr is already long going, could you please open an issue about that to be tackled as a refactoring task?